scripts/process_vad_data.py

```diff
@@ -0,0 +1,330 @@
+
+    training_list = [i for i in short_files if i not in exist_set]
+
+    with open(os.path.join(data_dir, "training_list.txt"), "w") as outfile:
+        outfile.write("\n".join(training_list))
+
+    logging.info(
+        f'Overall: {len(files)}, Train: {len(training_list)}, Validatoin: {len(validation_list)}, Test: {len(testing_list)}'
+    )
+
+
+def write_manifest(
+    data_dir,
+    out_dir,
+    files,
+    prefix,
+    manifest_name,
+    duration_stride=1.0,
+    duration_max=None,
+    filter_long=False,
+    duration_limit=10.0,
+):
+    seg_num = 0
+    skip_num = 0
+    if duration_max is None:
+        duration_max = 1e9
+
+    if not os.path.exists(out_dir):
+        print(f'outdir {out_dir} does not exist. Creat directory.')
+        os.mkdir(out_dir)
+
+    output_path = os.path.join(out_dir, manifest_name + '.json')
+    with open(output_path, 'w') as fout:
+        for file in files:
+
+            label = prefix
+
+            try:
+                x, _sr = librosa.load(file, sr=sr)
+                duration = librosa.get_duration(x, sr=sr)
+
+            except Exception:
+                continue
+
+            if filter_long and duration > duration_limit:
+                skip_num += 1
+                continue
+
+            offsets = []
+            durations = []
+
+            if duration > duration_max:
+                current_offset = 0.0
+
+                while current_offset < duration:
+                    difference = duration - current_offset
+                    segment_duration = min(duration_max, difference)
+
+                    offsets.append(current_offset)
+                    durations.append(segment_duration)
+
+                    current_offset += duration_stride
+
+            else:
+                offsets.append(0.0)
+                durations.append(duration)
+
+            for duration, offset in zip(durations, offsets):
+                metadata = {
+                    'audio_filepath': file,
+                    'duration': duration,
+                    'label': label,
+                    'text': '_',  # for compatibility with ASRAudioText
+                    'offset': offset,
+                }
+                json.dump(metadata, fout)
+                fout.write('\n')
+                fout.flush()
+                seg_num += 1
+    return skip_num, seg_num, output_path
+
+
+def load_list_write_manifest(data_dir, out_dir, filename, prefix, duration_stride=1.0, duration_max=1.0):
+
+    filename = prefix + '_' + filename
+    file_path = os.path.join(data_dir, filename)
+
+    with open(file_path, 'r') as allfile:
+        files = allfile.read().splitlines()
+
+    manifest_name = filename.split('_list.txt')[0] + '_manifest'
+    skip_num, seg_num, output_path = write_manifest(
+        data_dir,
+        out_dir,
+        files,
+        prefix,
+        manifest_name,
+        duration_stride,
+        duration_max,
+        filter_long=True,
+        duration_limit=100.0,
```

Comment:
duration_limit should be a user optional argument via argparse, whose default value is 100.

Reply:
If we set duration_limit to be an optional argument via argparse, we should also set duration stride and duration_max too, because they are changeable. But then the argparse will have too many argument I'm afraid. How about we let the argparse only cover high-level arguments?